Tests/OpenCombineTests/PublisherTests/ZipTests.swift

```diff
@@ -0,0 +1,692 @@
+
+        let zip = child1Publisher.zip(child2Publisher) { $0 + $1 }
+
+        var downstreamSubscription: Subscription?
+        let downstreamSubscriber = TrackingSubscriber(receiveSubscription: {
+            downstreamSubscription = $0
+            $0.request(.max(1))
+        })
+
+        zip.subscribe(downstreamSubscriber)
+
+        XCTAssertEqual(child1Publisher.send(100), .none)
+        XCTAssertEqual(child2Publisher.send(1), .none)
+        XCTAssertEqual(downstreamSubscriber.history, [.subscription("Zip"),
+                                                      .value(101)])
+
+        XCTAssertEqual(child1Publisher.send(200), .none)
+        XCTAssertEqual(child1Publisher.send(300), .none)
+        XCTAssertEqual(child2Publisher.send(2), .none)
+        // Surplus is sent downstream despite demand of zero
+        XCTAssertEqual(downstreamSubscriber.history, [.subscription("Zip"),
+                                                      .value(101),
+                                                      .value(202)])
+
+        XCTAssertEqual(child2Publisher.send(3), .none)
+        downstreamSubscription?.request(.max(1))
+        // Surplus is buffered for sending when demand resumes
+        XCTAssertEqual(downstreamSubscriber.history, [.subscription("Zip"),
+                                                      .value(101),
+                                                      .value(202),
+                                                      .value(303)])
+    }
+
+    private func getChildrenAndZipForArity(_ childCount: Int)
+        -> ([ChildInfo], AnyPublisher<Int, TestingError>)
+    {
+        var children = [ChildInfo]()
+        for _ in (0..<childCount) {
+            let subscription = CustomSubscription()
+            let publisher = CustomPublisher(subscription: subscription)
+            children.append(ChildInfo(subscription: subscription,
+                                      publisher: publisher))
+        }
+
+        let zip: AnyPublisher<Int, TestingError>
+
+        switch childCount {
+        case 2:
+            zip = AnyPublisher(children[0].publisher.zip(children[1].publisher)
+            { $0 + $1 })
+        case 3:
+            zip = AnyPublisher(children[0].publisher
+                .zip(children[1].publisher,
+                     children[2].publisher) { $0 + $1 + $2 })
+        case 4:
+            zip = AnyPublisher(children[0].publisher
+                .zip(children[1].publisher,
+                     children[2].publisher,
+                     children[3].publisher) { $0 + $1 + $2 + $3 })
+        default:
+            fatalError()
+        }
+
+        return (children, zip)
+    }
+
+    func testImmediateFinishWhenOneChildFinishesWithNoSurplus() {
+        ZipTests.arities.forEach { arity in
+            for childToFinish in (0..<arity) {
+                let description =
+                    String(format:"Zip\(arity) childToFinish=\(childToFinish)")
```

Comment:
Why use this initializer?
```suggestion
                let description = "Zip\(arity) childToFinish=\(childToFinish)"
```